definitions/features/hammer.rb

```diff
@@ -82,10 +82,12 @@ def on_invalid_host(custom_config)
   end
 
   def on_invalid_password(custom_config)
-    if !ready? && custom_config[:foreman][:password].nil?
+    if !ready? && custom_config[:foreman][:password] != password_from_answers(
```

Comment:
Instead of calling `password_from_answers` method twice, it would be good to catch value in a variable and use it.